lib/src/local_working_copy.rs

```diff
@@ -1267,10 +1289,14 @@ impl FileSnapshotter<'_> {
                 progress(&path);
             }
             if maybe_current_file_state.is_none()
-                && git_ignore.matches(path.as_internal_file_string())
+                && (git_ignore.matches(path.as_internal_file_string())
+                    || git_attributes
+                        .as_ref()
+                        .map(|ga| ga.matches(path.as_internal_file_string()))
+                        .unwrap_or(false))
```

Comment:
I realized that this conditional needs to be separate from `maybe_current_file_state.is_none()` (see https://github.com/jj-vcs/jj/pull/6035/files#r1997468647, which I left on the wrong PR)

Reply:
@bcspragu Had to do this also to actually have everything working:

https://github.com/jj-vcs/jj/compare/main...kejadlen:jj:lfs#diff-9654f3ae48bcd40ea4d2cdd4bfdebb9da27adb5373489816740d20642b174170R1435-R1441